central/networkpolicies/datastore/datastore_sac_test.go

```diff
@@ -0,0 +1,396 @@
+package store
+
+import (
+	"context"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/stackrox/rox/central/networkpolicies/datastore/internal/store"
+	boltStore "github.com/stackrox/rox/central/networkpolicies/datastore/internal/store/bolt"
+	pgdbStore "github.com/stackrox/rox/central/networkpolicies/datastore/internal/store/postgres"
+	"github.com/stackrox/rox/central/networkpolicies/datastore/internal/undodeploymentstore"
+	undodeploymentstoremock "github.com/stackrox/rox/central/networkpolicies/datastore/internal/undodeploymentstore/mocks"
+	"github.com/stackrox/rox/central/networkpolicies/datastore/internal/undostore"
+	undostoremock "github.com/stackrox/rox/central/networkpolicies/datastore/internal/undostore/mocks"
+	"github.com/stackrox/rox/central/role/resources"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/bolthelper"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/fixtures"
+	"github.com/stackrox/rox/pkg/postgres/pgtest"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/sac/testconsts"
+	"github.com/stackrox/rox/pkg/sac/testutils"
+	"github.com/stackrox/rox/pkg/uuid"
+	"github.com/stretchr/testify/suite"
+	bolt "go.etcd.io/bbolt"
+)
+
+func TestNetworkPolicySAC(t *testing.T) {
+	suite.Run(t, new(networkPolicySACSuite))
+}
+
+type networkPolicySACSuite struct {
+	suite.Suite
+
+	datastore DataStore
+
+	pool *pgxpool.Pool
+
+	engine *bolt.DB
+
+	storage             store.Store
+	undoStore           undostore.UndoStore
+	undoDeploymentStore undodeploymentstore.UndoDeploymentStore
+
+	mockCtrl *gomock.Controller
+
+	testContexts         map[string]context.Context
+	testNetworkPolicyIDs []string
+}
+
+func (s *networkPolicySACSuite) SetupSuite() {
+	var err error
+	if features.PostgresDatastore.Enabled() {
+		ctx := context.Background()
+		src := pgtest.GetConnectionString(s.T())
+		cfg, err := pgxpool.ParseConfig(src)
+		s.NoError(err)
+		s.pool, err = pgxpool.ConnectConfig(ctx, cfg)
+		s.NoError(err)
+		pgdbStore.Destroy(ctx, s.pool)
+		s.storage = pgdbStore.New(ctx, s.pool)
+	} else {
+		s.engine, err = bolthelper.NewTemp(s.T().Name() + ".db")
+		s.NoError(err)
+		s.storage = boltStore.New(s.engine)
+	}
```

Comment:
This pattern repeats in many places. Maybe we should extract it to a function

Reply:
I think there is a common piece, and another that is unique to each tested resource:

The common part is the extraction of the test postgres pool, and (in most stores) the extraction of the local rocks db. Cleanup of the test suite is common to SAC tests, I think. These could be extracted to common testutil functions.

Instantiation of the store and datastore is unique to each tested datastore.